Add java-imports

dakrone opened the pull request:
This is utility functions for adding `import` statements for Java. It
includes persistent caching of the packages for a class name so they
don't have to be re-entered every time.

purcell commented:
Thanks. Quick feedback:

- You're missing a `(provide ...)` form
- Every top-level symbol should have the `java-imports-` prefix, so symbols like `current-line-text` are not allowed
- You should add the missing package dependency on `(cl-lib "0.5")`

Cheers,
-Steve

dakrone commented:
Hi @purcell, thanks for taking a look. I changed all of the methods to begin with `java-imports-`, added `(provide ...)` form, and removed the `(require 'cl-lib)` statement entirely.

purcell commented:
Awesome, thanks.
